lib/asm-parser.js

```diff
@@ -293,6 +329,7 @@ class AsmParser extends AsmRegex {
         let inNvccCode = false;
 
         let inCustomAssembly = 0;
+        let lineNumber = 1;
```

Comment:
Instead of this, can you use the second parameter of the `forEach` ? on the line below I think:

```
asmLines.forEach((line, lineNumber0Indexed) => {
 ...
   labelDefinitions[match[1]] = lineNumber0Indexed + 1;
}
```

This is maybe about as clear (annoyingly it's off by one, being 0-indexed), but it saves any accidental forgetting to do `++lineNumber` (or any early returns etc). Means the two never get out of sync.

Reply:
Thank you for this. Unfortunately I forgot to decrement the value of the `lineNumber` variable when we remove item from the ast result :disappointed:
https://github.com/mattgodbolt/compiler-explorer/blob/4ddaac18c652411a6a7e841b2074b1105bc3cd24/lib/asm-parser.js#L324

Now I have refactored the code to use the length of the asm array. The code looks better :blush:

I also updated the test cases.